bin/i18n/i18n_script_utils.rb

```diff
@@ -210,4 +210,34 @@ def self.write_markdown_with_header(markdown, header, path)
   def self.sanitize_header!(header)
     header.slice!("title")
   end
+
+  # If a script is updated such that its destination directory changes after
+  # creation, we can end up in a situation in which we have multiple copies of
+  # the script file in the repo, which makes it difficult for the sync out to
+  # know which is the canonical version.
+  #
+  # To prevent that, here we proactively check for existing files in the
+  # filesystem with the same filename as our target script file, but a
+  # different directory. If found, we refuse to create the second such script
+  # file and notify of the attempt, so the issue can be manually resolved.
+  #
+  # Note we could try here to remove the old version of the file both from the
+  # filesystem and from github, but it would be significantly harder to also
+  # remove it from Crowdin.
+  def self.script_directory_change?(script_i18n_name, script_i18n_filename)
+    level_content_directory = "../#{I18N_SOURCE_DIR}/course_content"
+
+    matching_files = Dir.glob(File.join(level_content_directory, "**", script_i18n_name)).reject do |other_filename|
+      other_filename == script_i18n_filename
+    end
+
+    return false if matching_files.empty?
+
+    # Clean up the file paths, just to make our output a little nicer
+    base = Pathname.new(level_content_directory)
+    relative_matching = matching_files.map {|filename| Pathname.new(filename).relative_path_from(base)}
+    relative_new = Pathname.new(script_i18n_filename).relative_path_from(base)
+    STDERR.puts "Script #{script.name.inspect} wants to output strings to #{relative_new}, but #{relative_matching.join(' and ')} already exists"
```

Comment:
Would it be difficult or inappropriate to make this a HoneyBadger error instead? I debugged an issue recently and it turns out it was caused by the situation this function is checking for. However, no one on the team noticed this error message for a long time. It wasn't until someone from the i18n team asked why a particular file wasn't getting translation updates.

Any other suggestions / thoughts are welcome too.

Reply:
Not at all! HoneyBadger SGTM